drivers/hmc5883l/hmc5883l_saul.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_hmc5883l
+ * @brief       HMC5883L adaption to the RIOT actuator/sensor interface
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ */
+
+#include <string.h>
+
+#include "saul.h"
+#include "hmc5883l.h"
+
+static int read(const void *dev, phydat_t *res)
+{
+    hmc5883l_data_t data;
+    int ret = hmc5883l_read((const hmc5883l_t *)dev, &data);
+    if (ret < 0) {
+        return -ECANCELED;
+    }
+    res->val[0] = data.x / 1000;
+    res->val[1] = data.y / 1000;
+    res->val[2] = data.z / 1000;
+    res->unit = UNIT_GS;
+    res->scale = -3;
```

Comment:
The doc below states that `hmc5883l_read()` returns the values in micro Gauss (`uG`). So this should be `-6`, right?!

Reply:
Don't remember, I have to check again the datasheet.

Reply:
That should be correct. `hmc5883l_read` returns 32-bit values ​​in uGs. This divided by 1000 gives mGs. I decided to use 32-bit values ​​in `hmc5883l_read` not to lose accuracy when the raw data is multiplied by the gain.

Reply:
I decided to change everything to milli-Gauss (mGs) since the precission is far away from 1 mGs.

Reply:
I decided to change it back to uGs.